test/core/surface/channel_init_test.cc

```diff
@@ -37,10 +37,15 @@ const grpc_channel_filter* FilterNamed(const char* name) {
       new std::map<absl::string_view, const grpc_channel_filter*>;
   auto it = filters->find(name);
   if (it != filters->end()) return it->second;
+  static auto* name_factories =
+      new std::vector<std::unique_ptr<UniqueTypeName::Factory>>();
+  name_factories->emplace_back(std::make_unique<UniqueTypeName::Factory>(name));
+  auto unique_type_name = name_factories->back()->Create();
   return filters
-      ->emplace(name, new grpc_channel_filter{nullptr, nullptr, 0, nullptr,
-                                              nullptr, nullptr, 0, nullptr,
-                                              nullptr, nullptr, nullptr, name})
+      ->emplace(name,
+                new grpc_channel_filter{nullptr, nullptr, 0, nullptr, nullptr,
+                                        nullptr, 0, nullptr, nullptr, nullptr,
+                                        nullptr, unique_type_name})
```

Comment:
Can't just use `GRPC_UNIQUE_TYPED_NAME_HERE()`?

Reply:
No... that gets one unique name whenever it's typed.